tools/toolchains/arm.py

```diff
@@ -561,25 +603,64 @@ def is_not_supported_error(self, output):
     def parse_output(self, output):
         pass
 
+    def version_check(self):
+        ARM.version_check(self)
+
+        if self.is_mbed_studio_armc6:
+            # Current location of ARMC6 in the Mbed Studio installation is
+            # ./tools/ac6/bin. The license file is stored in ./tools, so the
+            # strategy to find the license file will be to keep checking each
+            # parent directory of ARMC6 for a file called "ac6-license.dat".
+
+            cur_path = TOOLCHAIN_PATHS["ARMC6"]
+            license_file = None
+            last_path = None
+            while cur_path != last_path:
+                head, tail = split(cur_path)
+                if (
+                    tail == "tools" and
+                    self.MBED_STUDIO_LICENSE_FILE_NAME in os.listdir(cur_path)
+                ):
+                    license_file = join(cur_path, self.MBED_STUDIO_LICENSE_FILE_NAME)
+                    break
+
+                else:
+                    last_path = cur_path
+                    cur_path = head
+
+            if license_file:
+                self.cmd_env = {
+                    "ARMLMD_LICENSE_FILE": license_file
+                }
+            else:
+                raise ToolException(
```

Comment:
This code doesn't work with Mbed Studio. Even when we pass `ARMLMD_LICENSE_FILE` we still end up with `else` statement.